sw/device/lib/runtime/ibex.h

```diff
@@ -0,0 +1,46 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_RUNTIME_IBEX_H_
+#define SW_DEVICE_LIB_RUNTIME_IBEX_H_
+
+#include <stddef.h>
+
+#include "sw/device/lib/base/stdasm.h"
+
+/**
+ * This header provides Ibex-specific functions, such as the clock frequency and
+ * cycle-accurate busy loops.
+ */
+
+/**
+ * The clock frequency of the Ibex core, in hertz.
+ */
+#ifdef SIMULATION
+static const size_t kIbexClockFreqHz = 500 * 1000;  // 500 kHz
+#else
+static const size_t kIbexClockFreqHz = 50 * 1000 * 1000;  // 50 MHz
+#endif
+
+/**
+ * Spins for roughly the number of cycles given.
+ *gs
+ * For best results, |cycles| should be a multiple of eight.
+ *
+ * This function should not be used for time-keeping.
+ *
+ * @arg cycles the number of cycles to burn.
+ */
+inline void ibex_busy_loop(size_t cycles) {
+  size_t out;  // Used to create an inout parameter below.
+  asm volatile(
+      "busy_loop%=:"
+      "  nop; nop; nop; nop;"    // 4 cycles.
+      "  addi %1, %1, -8;"       // 1 cycle.
+      "  bnez %1, busy_loop%=;"  // 3 cycles.
```

Comment:
This should be using `blez` here, or document that cycles __must_ be a multiple of 8. I'd prefer using blez.

Reply:
Done.